tests/system/src/test/java/test/robot/javafx/stage/AttributesTest.java

```diff
@@ -0,0 +1,187 @@
+import java.util.concurrent.TimeUnit;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import javafx.application.Platform;
+import javafx.scene.Scene;
+import javafx.stage.Stage;
+import javafx.stage.StageStyle;
+import javafx.scene.layout.Pane;
+import javafx.scene.paint.Color;
+import static junit.framework.Assert.assertFalse;
+import static junit.framework.Assert.assertTrue;
+import test.robot.testharness.VisualTestBase;
+
+import static test.util.Util.TIMEOUT;
+
+public class AttributesTest extends VisualTestBase {
+
+    private static final int WIDTH = 400;
+    private static final int HEIGHT = 400;
+
+    private static final Color BOTTOM_COLOR = Color.LIME;
+    private static final Color TOP_COLOR = Color.RED;
+
+    private static final double TOLERANCE = 0.07;
+
+    private Stage bottomStage;
+    private Stage topStage;
+
+    private void setupStages(boolean overlayed) throws InterruptedException {
+        final CountDownLatch bottomShownLatch = new CountDownLatch(1);
+        final CountDownLatch topShownLatch = new CountDownLatch(1);
+
+        runAndWait(() -> {
+            // Bottom stage, should be visible after top stage is iconified
+            bottomStage = getStage(false);
+            bottomStage.initStyle(StageStyle.DECORATED);
+            Scene bottomScene = new Scene(new Pane(), WIDTH, HEIGHT);
+            bottomScene.setFill(BOTTOM_COLOR);
+            bottomStage.setScene(bottomScene);
+            bottomStage.setX(0);
+            bottomStage.setY(0);
+            bottomStage.setOnShown(e -> Platform.runLater(bottomShownLatch::countDown));
+            bottomStage.show();
+        });
+
+        assertTrue("Timeout waiting for bottom stage to be shown",
+            bottomShownLatch.await(TIMEOUT, TimeUnit.MILLISECONDS));
+
+        runAndWait(() -> {
+            // Top stage, will be inconified
+            topStage = getStage(true);
+            topStage.initStyle(StageStyle.DECORATED);
+            Scene topScene = new Scene(new Pane(), WIDTH, HEIGHT);
+            topScene.setFill(TOP_COLOR);
+            topStage.setScene(topScene);
+            if (overlayed) {
+                topStage.setX(0);
+                topStage.setY(0);
+            } else {
+                topStage.setX(WIDTH);
+                topStage.setY(HEIGHT);
+            }
+            topStage.setOnShown(e -> Platform.runLater(topShownLatch::countDown));
+            topStage.show();
+        });
+
+        assertTrue("Timeout waiting for top stage to be shown",
+            topShownLatch.await(TIMEOUT, TimeUnit.MILLISECONDS));
+    }
+
+    public @Test void testIconifiedStage() throws InterruptedException {
+        setupStages(true);
+
+        runAndWait(() -> {
+            Color color = getColor(200, 200);
+            assertColorEquals(TOP_COLOR, color, TOLERANCE);
+
+            topStage.setIconified(true);
+        });
```

Comment:
This will show the stage before setting it as iconified. Testing correct behavior for a stage that is initially iconified/maximized/fullScreen prior to showing the stage is the main purpose of this test enhancement, so you will need to do something like split the creation and showing of the stages into separate methods or pass flags for the initial state of those three attributes as arguments to the setupStages method.

Reply:
The post-show behavior is already covered by other tests (e.g., IconifyTest). It wouldn't hurt to provide additional tests if easy, but it's not really necessary.